cpp/src/io/parquet/reader_impl.cu

```diff
@@ -630,10 +626,11 @@ class aggregate_reader_metadata {
     // Return true if column path is valid. e.g. if the path is {"struct1", "child1"}, then it is
     // valid if "struct1.child1" exists in this file's schema. If "struct1" exists but "child1" is
     // not a child of "struct1" then the function will return false for "struct1"
-    std::function<bool(column_name_info const*, int, std::vector<column_buffer>&)> build_column =
-      [&](column_name_info const* col_name_info,
-          int schema_idx,
-          std::vector<column_buffer>& out_col_array) {
+    std::function<bool(column_name_info const*, int, std::vector<column_buffer>&, bool)>
```

Comment:
Where does this function end? Should we extract it to a free function instead of a function object?

Reply:
I don't think it can be without some work.  The captures would make it messy.  Could be a followup PR I guess.